bin/ch/Helpers.cpp

```diff
@@ -4,6 +4,212 @@
 //-------------------------------------------------------------------------------------------------------
 #include "stdafx.h"
 
+//TODO: x-plat definitions
+#ifdef _WIN32
+typedef char16 TTDHostCharType;
+typedef struct _wfinddata_t TTDHostFileInfo;
+typedef intptr_t TTDHostFindHandle;
+typedef struct _stat TTDHostStatType;
+
+#define TTDHostPathSeparator _u("\\")
+#define TTDHostPathSeparatorChar _u('\\')
+#define TTDHostFindInvalid -1
+
+size_t TTDHostStringLength(const TTDHostCharType* str)
+{
+    return wcslen(str);
+}
+
+void TTDHostInitEmpty(TTDHostCharType* dst)
+{
+    dst[0] = _u('\0');
+}
+
+void TTDHostInitFromUriBytes(TTDHostCharType* dst, const byte* uriBytes, size_t uriBytesLength)
+{
+    memcpy_s(dst, MAX_PATH * sizeof(TTDHostCharType), uriBytes, uriBytesLength);
```

Comment:
(I may be out of context.) Using `MAX_PATH * ...` here completely bypasses the purpose of `memcpy_s`. If you want that, consider passing length of dst as a parameter?